testing/dart/path_test.dart

```diff
@@ -210,9 +210,11 @@ void main() {
     print(metrics);
     expect(metrics[0].length, 20);
     expect(metrics[0].isClosed, true);
-    expect(() => metrics[0].getTangentForOffset(4.0), throwsStateError);
-    expect(() => metrics[0].getTangentForOffset(4.0), throwsStateError);
+    expect(() => metrics[0].getTangentForOffset(4.0), isNotNull);
+    expect(() => metrics[0].extractPath(4.0, 10.0), isNotNull);
     expect(metrics[1].length, 10);
     expect(metrics[1].isClosed, false);
+    expect(() => metrics[1].getTangentForOffset(4.0), isNotNull);
+    expect(() => metrics[1].extractPath(4.0, 10.0), isNotNull);
```

Comment:
isNotNull is really really vague. We should test that it's actually returning the right thing.

Also we should check what happens when the path is mutated, since apparently that changed.

Reply:
It hasn't changed, it just wouldn't have worked at all before and now it's very likely undefined behavior (but perhaps still gets you the old values from before the path changed).  @reed-at-google - which is correct?

Reply:
I can update the test for getTangentForOffset to something more meaningful, but the getPath one gets a little tougher - I guess I could compute the metrics on it and make sure they come out to the expected length.

Reply:
The path is copied as soon as you create/reset the pathmeasure object, so there is no worry about edits to the path after that (including deleting it).

BTW: This is pretty cheap, as we just "ref" the path's internals, which perform a copy-on-write if someone actually edits the path after that. Thus in the common case, no actual copy has had to occur.